### PR TITLE
feat(ui): sort the additional data of Project/Component/Release follow alphabet order

### PR DESCRIPTION
**Pull request by quan-tv**
Signed-off-by: Tran Vu Quan <quan1.tranvu@toshiba.co.jp>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> * Issue: Additional data of Project/Component/Release are sorted by unclear algorithm
> * Solution: Sort the additional data of Project/Component/Release follow alphabet order (based on key values)

Issue: #1096 

 

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js [hunk 1]
```diff
@@ -120,28 +120,37 @@ define('modules/mergeWizard', [ 'jquery', 'modules/sw360Wizard' ], function($, s
...
         detailFormatter = detailFormatter || function(element) { return element; };
 
         result = $($.parseHTML('<fieldset id="' + normalizePropName(propName) + '" class="merge block">' +
                                '    <h4>' + propName + '</h4>' +
                                '</fieldset>'));
 
-        $.each(target, function(key, value) {
-            if (!source[key]) {
+        target.forEach((value, key) => {
+            if(!source.get(key)) {
```
Comment (ravi110336) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js:140]: Please check Indentation after if.

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java [hunk 2]
```diff
@@ -97,6 +97,24 @@ public boolean apply(String input) {
         return collection != null ? CASE_INSENSITIVE_ORDERING.immutableSortedCopy(collection) : ImmutableList.<String>of();
     }
 
+    /**
+     * Returns a sorted map containing the elements of the given unsorted map.
+     * The map is sorted follow the order.
+     */
+    public static Map<String, String> getSortedMap(Map<String, String> unsortedMap, boolean isAscending) {
+        if(unsortedMap == null && unsortedMap.size() <= 1) {
```
Comment (ravi110336) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java:105]: Please check indentation after if. same for line 109.

**Review by ravi110336 — COMMENTED** [review]
Reviewed and Tested the PR, Working fine. Please check for some minor indentation in code.

image

image

image

**Comment by quan-tv** [comment]
Hi @ravi110336,

For your comments related to indentation, I've checked the source code but could not find out which line got problem.
Could you please explain a little more?

Thank you very much!

https://github.com/eclipse/sw360/blob/ed49b2ec8396a4d16270b73c6457497a72396a38/frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js#L139-L151

https://github.com/eclipse/sw360/blob/ed49b2ec8396a4d16270b73c6457497a72396a38/libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java#L104-L116

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js (hunk 1 above)
Comment (ravi110336) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js:140]: @quan-tv ,We can add a space between the **if and (**    (please check the line 132) , its not an issue with the code but it is just the indentation we are maintaining.
like: if(!source.get(key)) {
 to   if (!source.get(key)) {

Comment (quan-tv) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js:140]: Ah, I got it. I myself also prefer this coding convention. I will update it.

Comment (quan-tv) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/mergeWizard.js]: @ravi110336 
The source code was updated!

**Comment by ravi110336** [comment]
Thanks @quan-tv , Code looks good to me, we can merge this PR.

**Pull request merged**